everware/home_handler.py

```diff
@@ -128,36 +130,40 @@ def get(self):
             ))
             return
 
-        if user.running and hasattr(user, "login_service") and user.login_service == "github":
-            if do_fork:
-                self.log.info('Will fork %s' % user.spawner.repo_url)
-                yield _fork_github_repo(
-                        user.spawner.repo_url,
-                        user.token,
-                    )
-                self.redirect('/hub/home')
-                return
-            if do_push:
-                self.log.info('Will push to fork')
-                yield _push_github_repo(
-                        user,
-                        user.spawner.repo_url,
-                        user.token,
-                    )
-                self.redirect('/hub/home')
-                return
-            repo_url = user.spawner.repo_url
-            fork_exists = yield _github_fork_exists(
-                                    user.name,
-                                    user.spawner.repo_url,
-                                    user.token,
-                                )
-            repository_changed = yield _repository_changed(user)
-        else:
-            repo_url = ''
-            fork_exists = False
-            repository_changed = False
-
+        branch_name = commit_sha = None
+        repo_url = ''
+        fork_exists = False
+        repository_changed = False
+        if user.running:
+            branch_name = user.spawner.branch_name
+            commit_sha = user.spawner.commit_sha
+            if getattr(user, "login_service", "") == "github":
```

Comment:
`getattr` is usually used to fetch a attribute specified by a variable `getattr(blah, foo)`.